set.mm

```diff
@@ -497748,6 +497788,159 @@ elements in a zero ring (see ~ 0ringirng ).  (Contributed by Thierry
       UVJYHUVLUVMVBXHZVCUVPYNYJUVRWMXIXJYFYIAYFYIUVNXKSXLXOXM $.
   $}
 
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Minimal polynomials
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c minPoly $. $( Minimal polynomial $)
+
+  $( Extend class notation with the minimal polynomial builder function. $)
+  cminply $a class minPoly $.
+
+  ${
+    $d e f p x $.
+    $( Define the minimal polynomial builder function .  (Contributed by
+       Thierry Arnoux, 19-Jan-2025.) $)
+    df-minply $a |- minPoly = ( e e. _V , f e. _V |-> ( x e. ( Base ` e )
+      |-> ( ( idlGen1p ` ( e |`s f ) ) ` { p e. dom ( e evalSub1 f )
+         | ( ( ( e evalSub1 f ) ` p ) ` x ) = ( 0g ` e ) } ) ) ) $.
+  $}
+
+  ${
+    $d B p q r $.  $d F q r $.  $d O p $.  $d P p q r $.  $d R q r $.
+    $d U p q r $.  $d X p $.  $d p ph q r $.
+    elvs1maprhm.q $e |- O = ( R evalSub1 S ) $.
+    elvs1maprhm.p $e |- P = ( Poly1 ` ( R |`s S ) ) $.
+    elvs1maprhm.b $e |- B = ( Base ` R ) $.
+    elvs1maprhm.u $e |- U = ( Base ` P ) $.
+    elvs1maprhm.r $e |- ( ph -> R e. CRing ) $.
+    elvs1maprhm.s $e |- ( ph -> S e. ( SubRing ` R ) ) $.
+    elvs1maprhm.y $e |- ( ph -> X e. B ) $.
+    elvs1maprhm.f $e |- F = ( p e. U |-> ( ( O ` p ) ` X ) ) $.
+    $( The function ` F ` mapping polynomials ` p ` to their subring evaluation
+       at a given point ` X ` is a ring homomorphism.  (Contributed by Thierry
+       Arnoux, 8-Feb-2025.) $)
+    elvs1maprhm $p |- ( ph -> F e. ( P RingHom R ) ) $=
```

Comment:
```suggestion
    evls1maprhm $p |- ( ph -> F e. ( P RingHom R ) ) $=
```

This proof is much shorter than I expected, I'll definitely look at it (and many other theorems) when it's generated.

Reply:
It's actually boilerplate, the function preserves addition and multiplication by `evls1addd` and `evls1muld`.
The image of the multiplicative unit polynomial (constant equal to 1) is trivially 1.

Reply:
(Might have missed the rename suggestion in my non-sequitur of a comment)

Reply:
Oh I did, thanks!